### ISSUE TITLE
[Bug]: Aborted due to failure, exceeded maximum size of 2693773 bytes / Server returned HTTP 404

### ISSUE THREAD

**Issue by seras42** [flatpak/flatpak#5648]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.15.4

### What Linux distribution are you using?

Fedora Linux

### Linux distribution version

Fedora Linux 38.20240107.0 (Kinoite)

### What architecture are you using?

x86_64

### How to reproduce

1. flatpak update
2. The update should fail

### Expected Behavior

1. flatpak update
2. Update was successful 

### Actual Behavior

```
flatpak update
Looking for updates…

        ID                                               Branch            Op            Remote            Download
 1. [✗] com.jetbrains.IntelliJ-IDEA-Community            stable            u             flathub           274,6 kB / 801,5 MB

Error: While pulling app/com.jetbrains.IntelliJ-IDEA-Community/x86_64/stable from remote flathub: Server returned HTTP 404
Updates complete.

```
Sometimes a different error is thrown such as

`Aborted due to failure (While pulling app/com.jetbrains.IntelliJ-IDEA-Community/x86_64/stable from remote flathub: URI https://dl.flathub.org/repo/deltas/hv/KajdCfFMGZEUBkZ2WtWKpO5Gs1UFe_09u7x5XM+vA-0_YJs8VP3PGCdjYnjk0ntr5eP9qCMCoQYZPFwPlmiNo/4 exceeded maximum size of 2693773 bytes)`

### Additional Information

The same issue has also been reported on the flathub/com.jetbrains.IntelliJ-IDEA-Community 
https://github.com/flathub/com.jetbrains.IntelliJ-IDEA-Community/issues/159#issue-2052762135

**Comment by ckujau** [comment]
Disabling delta updates appears to work as well:

```
flatpak update -vy --no-static-deltas com.jetbrains.IntelliJ-IDEA-Community
```
See [Unable to update com.jetbrains.IntelliJ-IDEA-Community](https://discourse.flathub.org/t/unable-to-update-com-jetbrains-intellij-idea-community/5880) for similar reports on this.

**Comment by seras42** [comment]
> Disabling delta updates appears to work as well:
> 
> ```
> flatpak update -vy --no-static-deltas com.jetbrains.IntelliJ-IDEA-Community
> ```
> 
> See [Unable to update com.jetbrains.IntelliJ-IDEA-Community](https://discourse.flathub.org/t/unable-to-update-com-jetbrains-intellij-idea-community/5880) for similar reports on this.

I really appreciate you sharing this solution, i think it might come in handy for anyone experiencing it. The distro i am in does automatic updates and for some reason one day the intelJ just updated successfully. I did not really do anything. Do you think i should close down this issue or leave it open for anyone else who might stumble upon it?

**Comment by ckujau** [comment]
This is a weird bug, "Server returned HTTP 404" suggests an error with Flathub and [#159: Error while updating to latest](https://github.com/flathub/com.jetbrains.IntelliJ-IDEA-Community/issues/159) looks like the correct place to report this. But then there's this "exceeded maximum size" message too and this I only found in the [flatpak/ppa-ostree repository](https://github.com/search?q=repo%3Aflatpak%2Fppa-ostree%20%22exceeded%20maximum%20size%22&type=code), which seems to have [zarro](https://www.bugzilla.org/about/zarroboogs.html) bugs, wow.

So, I'd keep this open and hope that someone from @flatpak/flatpak will comment here :-)

**Comment by GiulioTonellato** [comment]
I just had a very similar issue on SteamOS but with KDE Application Platform where I was getting those both errors (mostly 404 but also the maximum size one sometimes). Tried everything I was finding online and only this solved my problem once and for all.

> Disabling delta updates appears to work as well:
> 
> ```
> flatpak update -vy --no-static-deltas com.jetbrains.IntelliJ-IDEA-Community
> ```
> 
> See [Unable to update com.jetbrains.IntelliJ-IDEA-Community](https://discourse.flathub.org/t/unable-to-update-com-jetbrains-intellij-idea-community/5880) for similar reports on this.

At the beginning it was also causing other flatpaks not to update from the Discover app (they updated only through konsole after).
In my case the errors were:

Aborted due to failure (While pulling runtime/org.kde.Platform/x86_64/6.6 from remote flathub: Server returned HTTP 404)
or
Aborted due to failure (While pulling runtime/org.kde.Platform/x86_64/6.6 from remote flathub: URI https://dl.flathub.org/repo/deltas/Ix/B1UAgtrzsokqSg2yVDg4V4yogjQKdWuYi8XBYUVAw-4VthpO55ghIYWdBxLaw2c9LHnwmajaCCO45rG2B+_Rw/3 exceeded maximum size of 1430261 bytes)

solved with:
flatpak update -vy --no-static-deltas org.kde.Platform

Thank you!

**Comment by BBaoVanC** [comment]
This issue is now occurring on the LocalSend flatpak: flathub/org.localsend.localsend_app#84

**Comment by smcv** [comment]
> This issue is now occurring on the LocalSend flatpak: [flathub/org.localsend.localsend_app#84](https://github.com/flathub/org.localsend.localsend_app/issues/84)

No, that's a different error message, which seems to be caused by a libostree regression: https://github.com/ostreedev/ostree/issues/3635

**Comment by simonsan** [comment]
The fix still works, though: call flatpak with `flatpak update -y --no-static-deltas`.
